### PR TITLE
Improve documentation around ID Tokens

### PR DESCRIPTION
**Pull request by ajaska**
Includes a section on how to do caching, as part of #218 .

I've never used Sphinx for documentation before, and suggestions on clearer/more precise wording are always appreciated

### CODE REVIEW

File: google/oauth2/id_token.py [hunk 1]
```diff
@@ -12,7 +12,52 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Google ID Token helpers."""
+"""Google ID Token helpers.
+
+Provides support for verifying OAuth 2.0 ID Tokens, especially ones generated
```
Comment (jonparrott) [inline google/oauth2/id_token.py]: This are actual OpenID Connect ID Token. Might be worth linking to http://openid.net/specs/openid-connect-core-1_0.html#IDToken

File: google/oauth2/id_token.py [hunk 2]
```diff
@@ -12,7 +12,52 @@
...
+
+A general purpose ID Token verifier is available as :func:`verify_token`.
+
+Example::
+
+    from google.oauth2 import id_token
+    from google.auth.transport import requests
+
+    request = requests.Request()
+    try:
```
Comment (jonparrott) [inline google/oauth2/id_token.py]: nit: Please put a blank newline above control statements that start an indented block.

File: google/oauth2/id_token.py [hunk 3]
```diff
@@ -12,7 +12,52 @@
...
+        id_info = id_token.verify_oauth2_token(token, request, CLIENT_ID)
```
Comment (jonparrott) [inline google/oauth2/id_token.py]: Can you at least give an example client_id here?

File: google/oauth2/id_token.py [hunk 4]
```diff
@@ -12,7 +12,52 @@
...
+potential for network errors. This can be accomplished using an external
+library like `CacheControl`_ to create a cache-aware
+:class:`google.auth.transport.Request`::
+
+    from cachecontrol import CacheControl
+    from requests import session
+
+    from google.auth.transport import requests
+
+    sess = session()
```
Comment (jonparrott) [inline google/oauth2/id_token.py]: please use `session` (I'm very against abbreviation in names in Python, we're not Go programmers. :P)

File: google/oauth2/id_token.py (hunk 1 above)
Comment (ajaska) [reply google/oauth2/id_token.py]: Thanks, that's an important terminology distinction and useful link

File: google/oauth2/id_token.py (hunk 4 above)
Comment (ajaska) [reply google/oauth2/id_token.py]: I agree, was mostly trying to work around some funky name-spacing going on here (but can do that better)

Comment (ajaska) [reply google/oauth2/id_token.py]: How's this?

```diff
--- a/google/oauth2/id_token.py
+++ b/google/oauth2/id_token.py
@@ -26,9 +26,9 @@ A general purpose ID Token verifier is available as :func:`verify_token`.
 Example::
 
     from google.oauth2 import id_token
-    from google.auth.transport import requests
+    from google.auth.transport.requests
 
-    request = requests.Request()
+    request = google.auth.transport.requests.Request()
     try:
         id_info = id_token.verify_oauth2_token(token, request, CLIENT_ID)
 
@@ -48,13 +48,13 @@ library like `CacheControl`_ to create a cache-aware
 :class:`google.auth.transport.Request`::
 
     from cachecontrol import CacheControl
-    from requests import session
+    import requests
 
-    from google.auth.transport import requests
+    import google.auth.transport.requests
 
-    sess = session()
-    cached_sess = CacheControl(sess)
-    request = requests.Request(session=cached_sess)
+    session = requests.session()
+    cached_session = CacheControl(session)
+    request = google.auth.transport.requests.Request(session=cached_session)
 
 .. _CacheControl: https://cachecontrol.readthedocs.io
 """
```

Comment (jonparrott) [reply google/oauth2/id_token.py]: that'll work!

**Pull request merged**